frontend/catalyst/jax_tracer.py

```diff
@@ -533,12 +534,20 @@ def trace_to_jaxpr(func, static_argnums, abstracted_axes, args, kwargs):
     """
 
     with transient_jax_config({"jax_dynamic_shapes": True}):
-        with EvaluationContext(EvaluationMode.CLASSICAL_COMPILATION):
-            make_jaxpr_kwargs = {
-                "static_argnums": static_argnums,
-                "abstracted_axes": abstracted_axes,
-            }
-            jaxpr, out_type, out_treedef = make_jaxpr2(func, **make_jaxpr_kwargs)(*args, **kwargs)
+        make_jaxpr_kwargs = {
+            "static_argnums": static_argnums,
+            "abstracted_axes": abstracted_axes,
+        }
+        if pl_capture:
+            qml.capture.enable()
+            plxpr, out_type, out_treedef = make_jaxpr2(func, **make_jaxpr_kwargs)(*args, **kwargs)
+            qml.capture.disable()
+            jaxpr = catalyst.from_plxpr(plxpr)(*args, **kwargs)
+        else:
+            with EvaluationContext(EvaluationMode.CLASSICAL_COMPILATION):
+                jaxpr, out_type, out_treedef = make_jaxpr2(func, **make_jaxpr_kwargs)(
+                    *args, **kwargs
+                )
```

Comment:
By the way, we have a planned [`make_plxpr()`](https://app.shortcut.com/xanaduai/story/71823/a-make-plxpr-function-for-qnodes-captures-pennylane-with-control-flow) function coming down the pipeline, that should hopefully make this simpler (and will incorporate autograph as pre-processing).